src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -2384,3 +2388,366 @@ case class MatrixFilterEntries(child: MatrixIR, pred: IR) extends MatrixIR {
     mv.copy(rvd = newRVD)
   }
 }
+
+// follows key_by non-empty key
+case class TableAggregateByKey(child: TableIR, expr: IR) extends TableIR {
+  require(child.typ.keyOrEmpty.nonEmpty)
+
+  def children: IndexedSeq[BaseIR] = Array(child, expr)
+  
+  def copy(newChildren: IndexedSeq[BaseIR]): TableAggregateByKey = newChildren match {
+    case Seq(child: TableIR, _: IR) =>
+      TableAggregateByKey(child, expr)
+  }
+  
+  val typ: TableType = child.typ.copy(rowType = child.typ.keyType.get.merge(coerce[TStruct](expr.typ))._1)
+  
+  def execute(hc: HailContext): TableValue = {
+    val prev = child.execute(hc)
+    assert(prev.rvd.isInstanceOf[OrderedRVD])
```

Comment:
You don't need this line. The next one will do the same thing with `asInstanceOf`